BS440.py

```diff
@@ -91,44 +121,98 @@ def decodeWeight(handle, values):
+        11    tbw (percentage of water)
+        12    muscle (percentage of muscle) [0..100,0 %]       H (integer, length 2)
+        13    muscle (percentage of muscle)
+        14    bone (bone weight)            [0..100,0 %]       H (integer, length 2)
+        15    bone (bone weight)
+        --> Interpretation pattern:                            BIBBHHHHH
+    Notes: For kcal, fat, tbw, muscle, bone: First nibble = 0xf
     '''
     data = unpack('<BIBHHHHH', bytes(values[0:16]))
     retDict = {}
     retDict["valid"] = (data[0] == 0x6f)
     retDict["timestamp"] = sanitize_timestamp(data[1])
     retDict["person"] = data[2]
     retDict["kcal"] = data[3]
+    # Fat, water, muscle and bone are reported in *10. Hence, divide by 10.0
+    # To force results to be floats: devide by float.
     retDict["fat"] = (0x0fff & data[4])/10.0
     retDict["tbw"] = (0x0fff & data[5])/10.0
     retDict["muscle"] = (0x0fff & data[6])/10.0
     retDict["bone"] = (0x0fff & data[7])/10.0
     return retDict
 
+
+def sanitize_timestamp(timestamp):
+    '''
+    timestamp: timestamp of measurement transmitted by the scale
+
+    On some scales (e.g. BS410 and BS444, maybe others as well), time=0
+    equals 1/1/2010. However, goal is to have unix-timestamps. Thus, the
+    function converts the "scale-timestamp" to unix-timestamp by adding
+    the time-offset (most scales: 1262304000 = 01.01.2010) to the timestamp.
+    '''
+    retTS = 0
+    # Fail-safe: The timestamp will only be sanitized, if it will be
+    # below below the maximum unix timestamp (2147483647). Otherwise the
+    # non-sanitized timestamp will be taken.
+    if timestamp + time_offset < sys.maxsize:
+        retTS = timestamp + time_offset
+    else:
+        retTS = timestamp
+    # If already the non-sanitized timestamp is above the maximum unix timestamp
+    # 0 will be taken instead.
+    if timestamp >= sys.maxsize:
+        retTS = 0
+    return retTS
+
+
 def appendBmi(size, weightdata):
+    '''
+    appendBMI: Calculates the BMI (not calculated/ provided by the scale itself).
+
+    size: size of the person (fixed), stored in the scale
+    weightdata: list of weights of the previous x measurements
+    '''
     size = size / 100.00
     for element in weightdata:
-        element['bmi'] = round(element['weight'] / (size * size), 1)
+        '''
+        Some scales (e.g. BS 444) identify the max. 8 users via a "weight approximation".
+        If a new (or guest) user stands on the scale, the scale is not able to identify the user.
+        As a result, no size for the user is stored in the scale, and thus not transmitted
+        to the script. In these situations, the script stopped (division by zero). The following
+        check aims at solving this problem (https://github.com/keptenkurk/BS440/issues/102)
+        '''
+        if size != 0:
+            element['bmi'] = round(element['weight'] / (size * size), 1)
+        else:
+            element['bmi'] = 0
```

Comment:
You can simplify this a little
```suggestion
        if size == 0:
            element['bmi'] = 0
        else:
            element['bmi'] = round(element['weight'] / (size * size), 1)
```